File: backend/internal/models/member.go
package models

import "time"

// 团员发展阶段常量
const (
	StageApplicant       = "applicant"        // 入团申请人
	StageActivist        = "activist"         // 积极分子
	StageDevelopTarget   = "develop_target"   // 发展对象
	StagePoliticalReview = "political_review" // 政审备案
	StageLeagueMember    = "league_member"    // 正式团员
)

// MemberProfile 团员电子档案，覆盖团员发展全流程的核心字段
type MemberProfile struct {
	ID         uint   `gorm:"primarykey" json:"id"`
	UserID     uint   `gorm:"index;not null" json:"userId"`                   // 关联 users 表
	Name       string `gorm:"size:64;not null" json:"name"`                   // 姓名
	StudentNo  string `gorm:"uniqueIndex;size:32;not null" json:"studentNo"`  // 学号
	Gender     string `gorm:"size:8" json:"gender"`                           // 性别
	Birthday   string `gorm:"size:16" json:"birthday"`                        // 生日 yyyy-MM-dd
	IDCard     string `gorm:"size:32" json:"idCard"`                          // 身份证号
	Nation     string `gorm:"size:32" json:"nation"`                          // 民族
	Phone      string `gorm:"size:32" json:"phone"`                           // 联系电话
	College    string `gorm:"size:64" json:"college"`                         // 学院
	Major      string `gorm:"size:64" json:"major"`                           // 专业
	ClassName  string `gorm:"size:64" json:"className"`                       // 班级
	Stage      string `gorm:"size:32;index;default:applicant" json:"stage"`   // 当前阶段
	JoinDate   string `gorm:"size:16" json:"joinDate"`                        // 入团日期
	Remark     string `gorm:"type:text" json:"remark"`                        // 备注
	CreatedAt  time.Time `json:"createdAt"`
	UpdatedAt  time.Time `json:"updatedAt"`

	// 关联
	Applications     []LeagueApplication  `gorm:"foreignKey:ProfileID" json:"applications,omitempty"`
	ActivistRecords  []ActivistRecord     `gorm:"foreignKey:ProfileID" json:"activistRecords,omitempty"`
	DevelopRecords   []DevelopTargetRecord `gorm:"foreignKey:ProfileID" json:"developRecords,omitempty"`
	PoliticalRecords []PoliticalReview    `gorm:"foreignKey:ProfileID" json:"politicalRecords,omitempty"`
	Attachments      []MemberAttachment   `gorm:"foreignKey:ProfileID" json:"attachments,omitempty"`
}

// LeagueApplication 入团申请记录
type LeagueApplication struct {
	ID          uint      `gorm:"primarykey" json:"id"`
	ProfileID   uint      `gorm:"index;not null" json:"profileId"`
	ApplyDate   string    `gorm:"size:16;not null" json:"applyDate"`   // 申请日期
	Motivation  string    `gorm:"type:text" json:"motivation"`         // 入团动机
	Introducer  string    `gorm:"size:64" json:"introducer"`           // 介绍人
	Status      string    `gorm:"size:16;default:pending" json:"status"` // pending/approved/rejected
	ReviewNote  string    `gorm:"type:text" json:"reviewNote"`
	CreatedAt   time.Time `json:"createdAt"`
	UpdatedAt   time.Time `json:"updatedAt"`
}

// ActivistRecord 积极分子培养记录
type ActivistRecord struct {
	ID          uint      `gorm:"primarykey" json:"id"`
	ProfileID   uint      `gorm:"index;not null" json:"profileId"`
	StartDate   string    `gorm:"size:16;not null" json:"startDate"`
	Trainer     string    `gorm:"size:64" json:"trainer"`     // 培养联系人
	TrainPlan   string    `gorm:"type:text" json:"trainPlan"` // 培养计划
	Evaluation  string    `gorm:"type:text" json:"evaluation"`
	Score       float32   `json:"score"`
	CreatedAt   time.Time `json:"createdAt"`
	UpdatedAt   time.Time `json:"updatedAt"`
}

// DevelopTargetRecord 发展对象管理记录
type DevelopTargetRecord struct {
	ID            uint      `gorm:"primarykey" json:"id"`
	ProfileID     uint      `gorm:"index;not null" json:"profileId"`
	ConfirmedDate string    `gorm:"size:16;not null" json:"confirmedDate"`
	Mentor        string    `gorm:"size:64" json:"mentor"`
	PublicityNote string    `gorm:"type:text" json:"publicityNote"` // 公示说明
	Conclusion    string    `gorm:"size:255" json:"conclusion"`
	CreatedAt     time.Time `json:"createdAt"`
	UpdatedAt     time.Time `json:"updatedAt"`
}

// PoliticalReview 政审备案
type PoliticalReview struct {
	ID            uint      `gorm:"primarykey" json:"id"`
	ProfileID     uint      `gorm:"index;not null" json:"profileId"`
	ReviewDate    string    `gorm:"size:16;not null" json:"reviewDate"`
	Reviewer      string    `gorm:"size:64" json:"reviewer"`
	FamilyMembers string    `gorm:"type:text" json:"familyMembers"` // 直系亲属情况
	Conclusion    string    `gorm:"size:255" json:"conclusion"`
	Status        string    `gorm:"size:16;default:filed" json:"status"` // filed/recheck
	CreatedAt     time.Time `json:"createdAt"`
	UpdatedAt     time.Time `json:"updatedAt"`
}

// MemberAttachment 团员档案附件（存储在 MinIO 中）
type MemberAttachment struct {
	ID         uint      `gorm:"primarykey" json:"id"`
	ProfileID  uint      `gorm:"index;not null" json:"profileId"`
	Category   string    `gorm:"size:32;not null" json:"category"` // application/activist/develop/political/other
	FileName   string    `gorm:"size:255;not null" json:"fileName"`
	ObjectKey  string    `gorm:"size:255;not null" json:"objectKey"`
	URL        string    `gorm:"size:512" json:"url"`
	Size       int64     `json:"size"`
	UploadedBy uint      `json:"uploadedBy"`
	CreatedAt  time.Time `json:"createdAt"`
}

File: frontend/src/api/member.ts
import { useAuthStore } from '../stores/auth'

interface ApiResponse<T> {
  code: string
  message: string
  data: T
}

async function request<T>(url: string, init: RequestInit = {}): Promise<T> {
  const auth = useAuthStore()
  const headers = new Headers(init.headers)
  if (!headers.has('Content-Type') && !(init.body instanceof FormData)) {
    headers.set('Content-Type', 'application/json')
  }
  if (auth.token) headers.set('Authorization', `Bearer ${auth.token}`)

  const res = await fetch(url, { ...init, headers })
  const text = await res.text()
  const body = text ? (JSON.parse(text) as ApiResponse<T>) : ({} as ApiResponse<T>)
  if (!res.ok) {
    throw new Error(body.message || `请求失败 (${res.status})`)
  }
  return body.data
}

export interface MemberProfile {
  id: number
  userId: number
  name: string
  studentNo: string
  gender?: string
  birthday?: string
  idCard?: string
  nation?: string
  phone?: string
  college?: string
  major?: string
  className?: string
  stage: string
  joinDate?: string
  remark?: string
  applications?: LeagueApplication[]
  activistRecords?: ActivistRecord[]
  developRecords?: DevelopTargetRecord[]
  politicalRecords?: PoliticalReview[]
  attachments?: MemberAttachment[]
}

export interface LeagueApplication {
  id?: number
  applyDate: string
  motivation?: string
  introducer?: string
  status?: string
  reviewNote?: string
}

export interface ActivistRecord {
  id?: number
  startDate: string
  trainer?: string
  trainPlan?: string
  evaluation?: string
  score?: number
}

export interface DevelopTargetRecord {
  id?: number
  confirmedDate: string
  mentor?: string
  publicityNote?: string
  conclusion?: string
}

export interface PoliticalReview {
  id?: number
  reviewDate: string
  reviewer?: string
  familyMembers?: string
  conclusion?: string
  status?: string
}

export interface MemberAttachment {
  id: number
  category: string
  fileName: string
  url: string
  size: number
  createdAt: string
}

export interface MemberListResponse {
  items: MemberProfile[]
  total: number
  page: number
  size: number
}

export interface MemberArchive {
  profile: MemberProfile
  timeline: { date: string; stage: string; text: string }[]
  summary: Record<string, number | string>
}

export const memberApi = {
  list(params: { page?: number; size?: number; stage?: string; name?: string } = {}) {
    const search = new URLSearchParams()
    Object.entries(params).forEach(([k, v]) => {
      if (v !== undefined && v !== '') search.set(k, String(v))
    })
    return request<MemberListResponse>(`/api/v1/members?${search.toString()}`)
  },
  create(payload: Partial<MemberProfile>) {
    return request<MemberProfile>('/api/v1/members', {
      method: 'POST',
      body: JSON.stringify(payload),
    })
  },
  update(id: number, payload: Partial<MemberProfile>) {
    return request<MemberProfile>(`/api/v1/members/${id}`, {
      method: 'PUT',
      body: JSON.stringify(payload),
    })
  },
  detail(id: number) {
    return request<MemberProfile>(`/api/v1/members/${id}`)
  },
  remove(id: number) {
    return request<{ id: number }>(`/api/v1/members/${id}`, { method: 'DELETE' })
  },
  archive(id: number) {
    return request<MemberArchive>(`/api/v1/members/${id}/archive`)
  },
  addApplication(id: number, payload: LeagueApplication) {
    return request<LeagueApplication>(`/api/v1/members/${id}/applications`, {
      method: 'POST',
      body: JSON.stringify(payload),
    })
  },
  addActivist(id: number, payload: ActivistRecord) {
    return request<ActivistRecord>(`/api/v1/members/${id}/activists`, {
      method: 'POST',
      body: JSON.stringify(payload),
    })
  },
  addDevelop(id: number, payload: DevelopTargetRecord) {
    return request<DevelopTargetRecord>(`/api/v1/members/${id}/develop-targets`, {
      method: 'POST',
      body: JSON.stringify(payload),
    })
  },
  addPoliticalReview(id: number, payload: PoliticalReview) {
    return request<PoliticalReview>(`/api/v1/members/${id}/political-reviews`, {
      method: 'POST',
      body: JSON.stringify(payload),
    })
  },
  uploadAttachment(id: number, file: File, category: string) {
    const form = new FormData()
    form.append('file', file)
    form.append('category', category)
    return request<MemberAttachment>(`/api/v1/members/${id}/attachments`, {
      method: 'POST',
      body: form,
    })
  },
}

export const STAGE_OPTIONS = [
  { value: 'applicant', label: '入团申请人' },
  { value: 'activist', label: '积极分子' },
  { value: 'develop_target', label: '发展对象' },
  { value: 'political_review', label: '政审备案' },
  { value: 'league_member', label: '正式团员' },
]

export function stageLabel(value: string): string {
  return STAGE_OPTIONS.find((item) => item.value === value)?.label || value
}
